app/controllers/catalog.ts

```diff
@@ -103,4 +98,14 @@ export default class CatalogController extends Controller {
       .filterBy('featureIsProductWalkthrough')
       .rejectBy('isDismissed')[0] as FeatureSuggestionModel | null;
   }
+
+  shouldDisplayCourse(course: CourseModel) {
+    const isAlphaOrDeprecated = course.releaseStatusIsAlpha || course.releaseStatusIsDeprecated;
+    const isAuthorizedUser =
+      this.authenticator.currentUser && (this.authenticator.currentUser.isStaff || this.authenticator.currentUser.isCourseAuthor(course));
+    const hasUserProgress =
```

Comment:
```suggestion
    const userHasRepository =
```

Reply:
(don't introduce "new" names unless they actually simplify logic or compress things, sticking to existing names/concepts keeps things simpler)